src/utils/logging.ts

```diff
@@ -0,0 +1,48 @@
+export interface Logging {
+  debug(name: string, msg: string): void;
+  warn(name: string, msg: string): void;
+  error(name: string, msg: string): void;
+  info(name: string, msg: string): void;
+}
+
+export enum LogLevel {
+  Debug,
+  Info,
+  Warn,
+  Error,
+}
+
+export class ConsoleLogger implements Logging {
+  constructor(public currentLogLevel: LogLevel, private throwOnError: boolean) {
+    this.currentLogLevel = currentLogLevel;
+    this.throwOnError = throwOnError;
+  }
+
+  private aboveLogLevel(loglevel: LogLevel): boolean {
+    return loglevel >= this.currentLogLevel ? true : false;
+  }
+
+  private formatMessage(name: string, msg: string) {
+    const currenttime = new Date().toLocaleString();
+    return `${currenttime} [${name}] ${msg}`;
+  }
+
+  debug(name: string, msg: string): void {
+    if (this.aboveLogLevel(LogLevel.Debug)) console.debug(this.formatMessage(name, msg));
```

Comment:
I prefer to break it into multi-line if statement with curly braces:
```typescript
    if (this.aboveLogLevel(LogLevel.Debug)) {
      console.debug(this.formatMessage(name, msg));
    }
```

The reason is: if you later add something to this line and it becomes longer, prettier will break it into two lines. But it doesn't know to add curly braces, so you end up with something like:

```typescript
    if (this.aboveLogLevel(LogLevel.Debug))
      console.debug(this.formatMessage(name, msg) + 'some extra stuff so the line is longer');
```

Which has a [dangerous bug potential](https://embeddedgurus.com/barr-code/2014/03/apples-gotofail-ssl-security-bug-was-easily-preventable/).

Reply:
Same for warn() and info()

Reply:
Done